### DISCUSSION TITLE
Sometimes WA is Log out it self

### DISCUSSION THREAD

**Discussion by ryuaryuska** [aldinokemal/go-whatsapp-web-multidevice#218] [General]
sometimes the wa number is logout it self, so i created a health check service by curling the api /info. any suggest how the more  best practice approach?

**Comment by aldinokemal** [comment]
Polling `/app/info` will not tell you anything about the session. It only returns static server metadata (version, os, base_path, size limits) and still answers 200 when the device is fully logged out (`src/ui/rest/app.go:39`).

Use the status endpoints instead (both added in v8.0.0, commit `1b762b8`):

```bash
curl -s http://localhost:3000/devices/<device_id>/status
# {"results":{"is_connected":true,"is_logged_in":true,...}}

# or the header-scoped variant
curl -s -H "X-Device-Id: <device_id>" http://localhost:3000/app/status
```

`src/ui/rest/device.go:30` and `src/ui/rest/app.go:28`. The two flags need different handling:

- `is_logged_in: true` + `is_connected: false` = transient drop. `POST /devices/<device_id>/reconnect` (`src/ui/rest/device.go:29`).
- `is_logged_in: false` = the session was really unlinked (from the phone or by WhatsApp). No API call fixes that, you have to pair again via `GET /devices/<device_id>/login`.

For the transient case you do not need an aggressive poll. Auto-reconnect is already on (`src/infrastructure/whatsapp/device_manager.go:760`), there is a 5 minute liveness loop that calls `Connect()` when the client dropped (`src/ui/rest/helpers/common.go:29`), and every device is reconnected at boot (`src/ui/rest/helpers/common.go:13`).

The better approach for the real logout case is push instead of poll. On `events.LoggedOut` the server broadcasts over the websocket at `/ws`:

```json
{"code":"DEVICE_LOGGED_OUT","message":"Device logged out (slot kept)","result":{"device_id":"..."}}
```

`src/infrastructure/whatsapp/event_handler.go:222`. Subscribe to `/ws`, alert on that code, and keep a slow `/status` poll (5 minutes is plenty) as a fallback in case the websocket connection itself dies. There is currently no webhook event for logout, only the websocket broadcast, so if you want it delivered to an HTTP endpoint that is a welcome PR next to `handleLoggedOut`.

One cause worth ruling out if this happens often: `events.StreamReplaced` exits the process (`src/infrastructure/whatsapp/event_handler.go:290`). That fires when the same session is opened by a second client, so two containers pointed at the same session DB (`DBURI`, default `file:storages/whatsapp.db`, `src/config/settings.go:41`) will keep kicking each other out and it looks like a random self-logout. One session, one running instance.
